### PR TITLE
BaSP-TG: Some endpoints added to time-sheets

### PR DESCRIPTION
**Pull request by Martinzancocchia**
1. **Endpoints added:**
- `getByDate`
- `getByDesc`
- `add`
- `deleteById`

2. Routing added to /time-sheets/XXXX

3. Data added to /time-sheets.json

### CODE REVIEW

File: src/resources/time-sheets.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      res.status = 404;
+      res.send(`Error, time sheet with date= ${targetDate} not found.`);
+    }
+  } else {
+    res.status = 400;
+    res.send('Error, invalid request.');
+  }
+});
+
+router.get('/getByDesc', (req, res) => {
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:24]: Here the same, we miss the `/:description` to catch with the params.

File: src/resources/time-sheets.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  const queryDescription = req.query.description;
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:25]: We are using params to retrieve the data, we need to change de `.query` for `.params`

File: src/resources/time-sheets.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  if (queryDescription) {
+    const targetDesc = (queryDescription);
+    const targetTS = timeSheets.filter((current) => current.description === targetDesc).pop();
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:28]: is this `.pop` neccesary?

File: src/resources/time-sheets.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  fileSystem.writeFile('src/data/time-sheets.json', JSON.stringify(timeSheets, null, 4), (err) => {
+    if (err) {
+      res.send('Cannot save new user');
+    } else {
+      res.send(`User created\n ${JSON.stringify(newTimeSheet)}`);
+    }
+  });
+});
+
+router.delete('/deleteById', (req, res) => {
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:53]: we need to add `/:id`

File: src/resources/time-sheets.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  const timeSheetsId = parseInt(req.query.id, 10);
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:54]: change query for params

File: src/resources/time-sheets.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  const timeSheetsFilter = timeSheets.filter((timeSheet) => timeSheet.id !== timeSheetsId);
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:55]: we need to parse to number the `timeSheetsId` in orden to make the delete method functional

File: src/resources/time-sheets.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  fileSystem.writeFile('src/data/time-sheets.json', JSON.stringify(timeSheetsFilter, null, '\n'), (err) => {
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:56]: is no neccesary the second and third parameters in JSON.stringify

File: src/resources/time-sheets.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
+const express = require('express');
+const fileSystem = require('fs');
+const timeSheets = require('../data/time-sheets.json');
+
+const router = express.Router();
+
+router.get('/getByDate', (req, res) => {
+  const queryDate = req.query.date;
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:8]: We need to replace query for params, we are working with params

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/resources/time-sheets.js (hunk 6 above)
Comment (Martinzancocchia) [reply src/resources/time-sheets.js:55]: Hola Ayom, ya esta parseado, a que te referis?

File: src/resources/time-sheets.js (hunk 8 above)
Comment (Martinzancocchia) [reply src/resources/time-sheets.js:8]: Buenas ayom, podemos dejar la fecha con query paramas, por el tema de las barras "/"

Comment (ignacioaristo) [reply src/resources/time-sheets.js]: Ok, no problem in this case.

File: src/resources/time-sheets.js (hunk 6 above)
Comment (ignacioaristo) [reply src/resources/time-sheets.js]: Right now the delete function is not working, if we remove the parseInt will work.

File: src/resources/time-sheets.js (hunk 7 above)
Comment (ignacioaristo) [reply src/resources/time-sheets.js]: Lets keep it anyway

File: src/index.js [hunk 9]
```diff
@@ -2,23 +2,18 @@
...
 
 // use "require" to import JSON files
-const admins = require('./data/admins.json');
+const timeSheetsRouter = require('./resources/time-sheets');
 
 const app = express();
 const port = process.env.PORT || 3000;
 
 app.use(express.json());
+app.use('/time-sheets', timeSheetsRouter);
```
[the remark this answers is not in the document]
Comment (Martinzancocchia) [reply src/index.js]: done

File: src/resources/time-sheets.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
+const express = require('express');
+const fileSystem = require('fs');
+const timeSheets = require('../data/time-sheets.json');
+
+const router = express.Router();
+
+router.get('/getByDate', (req, res) => {
```
[the remark this answers is not in the document]
Comment (Martinzancocchia) [reply src/resources/time-sheets.js:7]: fixed

File: src/resources/time-sheets.js (hunk 1 above)
Comment (Martinzancocchia) [reply src/resources/time-sheets.js]: fixed!!!

File: src/resources/time-sheets.js (hunk 2 above)
Comment (Martinzancocchia) [reply src/resources/time-sheets.js]: done!

File: src/resources/time-sheets.js (hunk 3 above)
Comment (Martinzancocchia) [reply src/resources/time-sheets.js]: .pop deleted

File: src/resources/time-sheets.js (hunk 4 above)
Comment (Martinzancocchia) [reply src/resources/time-sheets.js]: added

File: src/resources/time-sheets.js (hunk 5 above)
Comment (Martinzancocchia) [reply src/resources/time-sheets.js]: done

File: src/resources/time-sheets.js (hunk 7 above)
Comment (Martinzancocchia) [reply src/resources/time-sheets.js]: that can stay, fabri has talked about it with you

**Pull request merged**